### ISSUE TITLE
Creating task - Improvement

### ISSUE THREAD

**Issue by fiuzaB** [johannesjo/super-productivity#219]
### Your Environment
* Version used: 
* Operating System and version: Windows 10

### Expected Behavior
There should be a button to create a new task. Mayble close to the "Ready to Work". I spend some time trying to figure out how to add a task. It's awesome to be able to just tap enter and the task is added but not very intuitive (of course, that´s my opinion )

### Current Behavior
Today is hard to know what to do to add a new task. If people are like me, they will hit the "Ready to Work" button and expect the task to be created.

**Comment by johannesjo** [comment]
You're absolutely right! This needs to be improved. Will require quite some changes, though. I'll work on that as soon as i can.

**Comment by fiuzaB** [comment]
If you need any help, count me in.

**Comment by johannesjo** [comment]
Oh that's great to hear!

My general idea is to redo the whole ui for the add task use case using a combination of todoists approach of a split input bar (one input for the title, one for the due date of the task) and the general design of microsoft todo: an empty input at the buttom of the list, which is always visible on the task list page. 

The most complex part in respect to the code required is probably the date selection part and in place of a seperate input we also could extend the short syntax (though a separate input is probably easier for new users). But we can always add the whole due date selection later on when we're done with the general ui.

There are several tricky bits to consider about the ui as well. One of it is the how to deal with the toast notifications. The material design spec specifies for fab buttons (the big ones which are usually used in the bottom center or bottom right) to below the content (https://material.io/components/snackbars/#placement). Maybe a always visible task input is a little bit similiar. Either way we need some solution for what happens when a snack notification appears. Having it appear just on top of the input is probably not a good idea.

The other thing which comes to mind is the current backlog implementation. What happens when the backlog is pulled up? Would the input bar move up with it as well? Or do we let it disappear completely. My first feeling is, that this might be the best solution for now. 

Don't know. What do you think? I've to admit, that just adding a button might be the easier solution :D

**Comment by fiuzaB** [comment]
> 
> 
> Oh that's great to hear!
> 
> My general idea is to redo the whole ui for the add task use case using a combination of todoists approach of a split input bar (one input for the title, one for the due date of the task) and the general design of microsoft todo: an empty input at the buttom of the list, which is always visible on the task list page.
> 
I got the part of the todois approach but I didn't understand very well the empty input at the bottom. I mean, we have a lot of space. If we set an element at the bottom of the screen, we will have a big empty space in between the first input and the last, right ? The reason you want to have this "bottom input" is to minimize the amount of clicks the user has to do before inserting a new task ? 
> The most complex part in respect to the code required is probably the date selection part and in place of a seperate input we also could extend the short syntax (though a separate input is probably easier for new users). But we can always add the whole due date selection later on when we're done with the general ui.
> 
Yeah, date is always a problem. Checking the system, I got two types of calendars. One when doing the schedule thing and the other going through Estimate Time > Add Time Spent for other day. Maybe one type of calendar would make life easier too. 
> There are several tricky bits to consider about the ui as well. One of it is the how to deal with the toast notifications. The material design spec specifies for fab buttons (the big ones which are usually used in the bottom center or bottom right) to below the content (https://material.io/components/snackbars/#placement). Maybe a always visible task input is a little bit similiar. Either way we need some solution for what happens when a snack notification appears. Having it appear just on top of the input is probably not a good idea.
> 
Yep. That would be annoying. I'm not 100% familiar with the system. Do you have an idea when the snackbar appears ? Probably after an user interaction, right ?
> The other thing which comes to mind is the current backlog implementation. What happens when the backlog is pulled up? Would the input bar move up with it as well? Or do we let it disappear completely. My first feeling is, that letting it disappear (maybe in favor of a static variant at the end of the list) might be the best solution for now.
> 
Yep. I agree. 
> Well and then there is the add task overlay... Maybe we can just use the same one we use for the task list, but slide it up, when being summoned from somewhere else. Not really sure...
> 
I like the way it is today. It is simple. I don't know if it was your intention, but maybe we could have a more detailed way to add tasks when the user come from the Task Menu and a simple one when coming from the Plus button. I see it like a shortcut, a fast way to add a new task.
> Don't know. What do you think? I've to admit, that just adding a button might be the easier solution :D
Waaay easier. It always depends on time. But we could do something to fix this issue for now and open new issues to deal with these details.
Another option is to insert the task if the input is different than empty and the user clicks on "Ready to Work" .

**Comment by johannesjo** [comment]
Thanks for the feedback! :) Maybe let's go with a simple button for now. 

This is what I mean by the bar at the bottom.
![image](https://user-images.githubusercontent.com/1456265/65711954-480c6980-e096-11e9-8ac7-80f45c8c69fa.png)

![image](https://user-images.githubusercontent.com/1456265/65712135-a3d6f280-e096-11e9-8ecb-4480b4d8e46a.png)

**Comment by KonTy** [comment]
> The most complex part in respect to the code required is probably the date selection part and in place of a s

We could also use this opportunity to do some natural language processing using SugarJS library
https://sugarjs.com/

I love the Idea just typing "My todo due @tomorrow" and the it will be added automatically. This will save so many clicks.

**Comment by KonTy** [comment]
Oh man I wish I had more time for this, there is so much we can do with your app! So many cool ideas floating around.

**Comment by fiuzaB** [comment]
I'll take this, if that is alright.

**Comment by fiuzaB** [comment]
Something like this ?

![like_this](https://user-images.githubusercontent.com/11810770/65994348-7957b180-e469-11e9-9fb5-40e52ddaffaf.PNG)

Or ( I like this one ):
![or_like_this](https://user-images.githubusercontent.com/11810770/65994390-8bd1eb00-e469-11e9-95a9-bd46ef2a7f9a.PNG)

**Comment by KonTy** [comment]
Personally for me looking at how the other Todo apps behave you are sort of expected to hit enter even on the phone, because even on the phone you are typing at this time and you have the keyboard up and there is an Enter key there. IMHO having an extra button adds unnecessary clutter, and it is just a matter of a learning curve to know that you have to hit enter. 

If you still decided to add a button, one thing to consider is to change the icon on the left side to a SAVE icon. That's how MS Todo does it for example:

![Task Mockup](https://user-images.githubusercontent.com/9524513/65996045-82d22100-e44b-11e9-9123-b8ec66c4c617.jpg)

**Comment by johannesjo** [comment]
@KonTy nice suggestion. Could you maybe open up a new issue about the short syntax? I probably wouldn't import the whole sugar.js lib. It's most of the time much more performant to code those things for the specific case at hand, but the date stuff is pretty nice!

@fiuzaB  Sorry, I think I misunderstood you. I thought you meant another button to show the add task bar (Shift+A or the + icon on the top).  I have to agree with Tony. Hitting enter is not too uncommon for search bars and single inputs of all kinds. The only way I can imagine this, is a very subtle approach like the one KonTy mentioned or like this one:

![Screenshot from 2019-10-01 22-02-16](https://user-images.githubusercontent.com/1456265/65996427-aae77180-e497-11e9-9344-32130fa55780.png)

What might be confusing is the button below. Maybe we need to give it a little bit more margin to the input?

![Uploading Screenshot from 2019-10-01 22-09-24.png…]()

**Comment by fiuzaB** [comment]
> 
> 
> Personally for me looking at how the other Todo apps behave you are sort of expected to hit enter even on the phone, because even on the phone you are typing at this time and you have the keyboard up and there is an Enter key there. IMHO having an extra button adds unnecessary clutter, and it is just a matter of a learning curve to know that you have to hit enter.
> 
> If you still decided to add a button, one thing to consider is to change the icon on the left side to a SAVE icon. That's how MS Todo does it for example:
> 
> ![Task Mockup](https://user-images.githubusercontent.com/9524513/65996045-82d22100-e44b-11e9-9123-b8ec66c4c617.jpg)

Way better. I just checked Microsoft Todo : 
![image](https://user-images.githubusercontent.com/11810770/65996974-dace4f00-e46e-11e9-961d-a2adcdd8f2a2.png) its way more classy.




> 
> 

> @fiuzaB Sorry, I think I misunderstood you. I thought you meant another button to show the add task bar (Shift+A or the + icon on the top). I have to agree with Tony. Hitting enter is not too uncommon for search bars and single inputs of all kinds. The only way I can imagine this, is a very subtle approach like the one KonTy mentioned or like this one:
> 
> ![Screenshot from 2019-10-01 22-02-16](https://user-images.githubusercontent.com/1456265/65996427-aae77180-e497-11e9-9344-32130fa55780.png)
> 
> What might be confusing is the button below. Maybe we need to give it a little bit more margin to the input?
> 
> ![Screenshot from 2019-10-01 22-09-24](https://user-images.githubusercontent.com/1456265/65996715-3c56e380-e498-11e9-98ea-dff92ee7fa2f.png)

No problem ! I'm using your app in a Windows machine and I'm used to hit buttons while working on Windows. So I tought if I clicked Ready To Work it would add my task to the queue :)

How about we close this task and if ppl start talking about it we open this issue again ?

**Comment by johannesjo** [comment]
@fiuzaB githubs issue template is pretty good now. So people should find it when they try to create a new one. I will add some more space to the read to work button. You're right that this a little confusing at the moment.

**Comment by KonTy** [comment]
> @KonTy nice suggestion. Could you maybe open up a new issue about the short syntax? I probably wouldn't import the whole sugar.js lib.

Will do

**Comment by johannesjo** [comment]
@KonTy that really looks promising! 

One request though: Could you maybe try to keep stuff more separated and focused on the issue titles and open up new issues for new subjects? I think it gets hard to follow for people which are looking for a solution described in the title. :D

For a less focused general discussion or just random ideas, I also opened up this spectrum chat:
https://spectrum.chat/super-productivity

**Comment by KonTy** [comment]
> One request though: Could you maybe try to keep stuff more separated and focused on the issue titles and open up new issues for new subjects? I think it gets hard to follow for people which are looking for a solution described in the title. :D

Sorry about that. I didn't mean to randomize the thread. I just get too excited about the potential of this app. It is so close to being the *best* productivity/todo app out there.

**Comment by johannesjo** [comment]
@KonTy no worries :)

**Comment by johannesjo** [comment]
With the addition of advanced short syntax this can be closed I think. Please feel free to open up a new issue, if you like to see further improvements.
